src/components/application_manager/src/commands/mobile/create_interaction_choice_set_request.cc

```diff
@@ -108,42 +142,63 @@ void CreateInteractionChoiceSetRequest::Run() {
     return;
   }
 
-  Result::eType result = CheckChoiceSet(app);
+  Result::eType result = CheckChoiceSet();
   if (Result::SUCCESS != result) {
     SendResponse(false, result);
     return;
   }
   uint32_t grammar_id = application_manager_.GenerateGrammarID();
   (*message_)[strings::msg_params][strings::grammar_id] = grammar_id;
-  app->AddChoiceSet(choice_set_id_, (*message_)[strings::msg_params]);
   SendVRAddCommandRequests(app);
+  app->AddChoiceSet(choice_set_id_, (*message_)[strings::msg_params]);
 }
 
-mobile_apis::Result::eType CreateInteractionChoiceSetRequest::CheckChoiceSet(
-    ApplicationConstSharedPtr app) {
-  using namespace smart_objects;
+mobile_apis::Result::eType CreateInteractionChoiceSetRequest::CheckChoiceSet() {
+  using smart_objects::SmartObject;
+  using smart_objects::SmartArray;
   LOG4CXX_AUTO_TRACE(logger_);
 
   std::set<uint32_t> choice_id_set;
+  UniqueParamChecker choice_menu_name_checker(strings::menu_name);
+  UniqueParamChecker choice_vr_command_checker(strings::vr_commands);
 
   const SmartArray* choice_set =
       (*message_)[strings::msg_params][strings::choice_set].asArray();
 
   SmartArray::const_iterator choice_set_it = choice_set->begin();
 
   for (; choice_set->end() != choice_set_it; ++choice_set_it) {
-    std::pair<std::set<uint32_t>::iterator, bool> ins_res =
-        choice_id_set.insert((*choice_set_it)[strings::choice_id].asInt());
-    if (!ins_res.second) {
+    const uint32_t choice_id = (*choice_set_it)[strings::choice_id].asUInt();
+    if (!utils::InsertIntoSet(choice_id, choice_id_set)) {
       LOG4CXX_ERROR(logger_,
-                    "Choise with ID "
-                        << (*choice_set_it)[strings::choice_id].asInt()
-                        << " already exists");
+                    "Choise with ID " << choice_id << " already exists");
```

Comment:
@wolfylambova22, actually it should be `Choice ID`. It's ID of a `Choice`, not a chosen ID